src/bin.js

```diff
@@ -2,29 +2,124 @@
 
 'use strict'
 
-const signalling = require('./index')
-const argv = require('minimist')(process.argv.slice(2))
+/* eslint-disable no-console */
 
-let server
+const Server = require('./')
+const Libp2p = require('libp2p')
 
-/* eslint-disable no-console */
+const WS = require('libp2p-websockets')
+const TCP = require('libp2p-tcp')
+const MULTIPLEX = require('libp2p-multiplex')
+const SPDY = require('libp2p-spdy')
+const SECIO = require('libp2p-secio')
+
+const inDocker = Boolean(process.env.DOCKERMODE)
+
+const path = require('path')
+const fs = require('fs')
+
+const Peer = require('peer-info')
+const Id = require('peer-id')
 
-signalling.start({
+const argv = require('minimist')(process.argv.slice(2))
+
+const config = {
   port: argv.port || argv.p || process.env.PORT || 9090,
   host: argv.host || argv.h || process.env.HOST || '0.0.0.0',
-  cryptoChallenge: !(argv.disableCryptoChallenge || process.env.DISABLE_CRYPTO_CHALLENGE),
-  strictMultiaddr: !(argv.disableStrictMultiaddr || process.env.DISABLE_STRICT_MULTIADDR),
-  metrics: !(argv.disableMetrics || process.env.DISABLE_METRICS)
-}, (err, _server) => {
-  if (err) { throw err }
-  server = _server
-
-  console.log('Listening on:', server.info.uri)
-})
+  genconf: argv.genconf || argv.g || process.env.GENCONF || false,
+  metrics: false, // !(argv.disableMetrics || process.env.DISABLE_METRICS)
```

Comment:
Should be environment variable as well.

Reply:
It's currently disabled as metrics aren't implemented yet